### DISCUSSION TITLE
Object store get streaming data?

### DISCUSSION THREAD

**Discussion by tamis-laan** [nats-io/nats.py#590] [Q&A]
I want to put and get large files into the object store that don't fit into memory. From the put signature I can see I can use a `BufferedIOBase` allowing me to stream the data into the store:

``` python
(method) def put(
    name: str,
    data: str | bytes | BufferedIOBase,
    meta: ObjectMeta | None = None
) -> Coroutine[Any, Any, ObjectInfo]
```

However it looks like I can't get the data out in a streaming manner:
``` python
audio = await bucket.get(str(audio_id))
print(type(audio), type(audio.data))
```

```
<class 'nats.js.object_store.ObjectStore.ObjectResult'> <class 'bytes'>
```

As you can see the `audio.data` property is of type `bytes` and is completely loaded into RAM memory.

**Comment by wallyqs** [comment]
here is an example https://github.com/nats-io/nats.py/blob/d6f4f97dc646d7a7c7cf1322f7aac899d4ddc6bf/tests/test_js.py#L3413

**Comment by tamis-laan** [comment]
@wallyqs 

Thnx! I was so focused on the return object I didn't see this option
